src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -60,19 +60,21 @@ type FlowNode = TFlowNode | TActivity | TReceiveTask | TEventBasedGateway | TTex
 type BpmnSemanticType = keyof TProcess;
 
 const computeSubProcessKind = (processedSemanticType: BpmnSemanticType, bpmnElement: TSubProcess): ShapeBpmnSubProcessKind => {
-  if (processedSemanticType == 'transaction') {
+  if (processedSemanticType == 'adHocSubProcess') {
```

Comment:
**thought:** I also considered using "switch" instead of "if/else", but I wasn't convinced it was a better syntax.

```ts
  switch (processedSemanticType) {
    case 'adHocSubProcess':
      return ShapeBpmnSubProcessKind.AD_HOC;
    case 'transaction':
      return ShapeBpmnSubProcessKind.TRANSACTION;
    default:
      return !bpmnElement.triggeredByEvent ? ShapeBpmnSubProcessKind.EMBEDDED : ShapeBpmnSubProcessKind.EVENT;
  }
```

Reply:
I don't have a strong preference, but personally, I find the **switch** syntax to be slightly more readable in this context.

Reply:
Okay, I'll move on to the "switch" implementation 😸

Reply:
Done in f3a41c6a